### PR TITLE
Return empty result when YTS api gives malformed response. Resolves #11040

### LINKED ISSUE #11040: "Spaces" in Title search for YTS breaks results and causes Radarr to stop searching.
Hi,

I'm running Jackett on Unraid with Radarr.

When i run a search with a space in the title i get the following error:-

`<error code="900" description="Jackett.Common.IndexerException: Exception (yts): API error, movies missing ---> System.Exception: Parse error ---> System.Exception: API error, movies missing at Jackett.Common.Indexers.YTS.PerformQuery(TorznabQuery query, Int32 attempts) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/YTS.cs:line 138 --- End of inner exception stack trace --- at Jackett.Common.Indexers.BaseWebIndexer.OnParseError(String results, Exception ex) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 682 at Jackett.Common.Indexers.YTS.PerformQuery(TorznabQuery query, Int32 attempts) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/YTS.cs:line 210 at Jackett.Common.Indexers.YTS.PerformQuery(TorznabQuery query) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/YTS.cs:line 87 at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query, Boolean isMetaIndexer) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 390 --- End of inner exception stack trace --- at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query, Boolean isMetaIndexer) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 390 at Jackett.Common.Indexers.BaseWebIndexer.ResultsForQuery(TorznabQuery query, Boolean isMetaIndexer) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 659 at Jackett.Server.Controllers.ResultsController.Torznab(TorznabRequest request) in /home/vsts/work/1/s/src/Jackett.Server/Controllers/ResultsController.cs:line 427"/>`

Its probably YTS but if you search for a movie that isnt availabale with a space in the title is comes back with an error, which in turn breaks Radarr and stops it from continuing its search, so if you are looking for 25+ movies it will stop the search once this error is recieved.

And exmaple API serach is as follows:-

http://192.168.1.44:9117/api/v2.0/indexers/yts/results/torznab/api?t=movie&cat=2000,2010,2020,2030,2035,2040,2045,2050,2060&extended=1&apikey=(API Key)&offset=0&limit=100&q=Land2021

The above will come back with a proper response (Even though the movie is missing)

<?xml version="1.0" encoding="UTF-8"?>
<rss version="1.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:torznab="http://torznab.com/schemas/2015/feed">
  <channel>
    <atom:link href="http://192.168.1.44:9117/" rel="self" type="application/rss+xml" />
    <title>YTS</title>
    <description>YTS is a Public torrent site specialising in HD movies of small size</description>
    <link>https://yts.mx/</link>
    <language>en-us</language>
    <category>search</category>
  </channel>
</rss>

Where are this:-

http://192.168.1.44:9117/api/v2.0/indexers/yts/results/torznab/api?t=movie&cat=2000,2010,2020,2030,2035,2040,2045,2050,2060&extended=1&apikey=(API Key)&offset=0&limit=100&q=Land%202021

Will cause the above error (Due to the space between Land 2021)

I hope this makes sense ??

Thanks again for your amazing app !

### PR DESCRIPTION
**Pull request by XYZJR**
#11040

### CODE REVIEW

**Comment by ilike2burnthing** [comment]
Tested, working.

Possibly the error wording needs tweaked though, or preferably not relying on `movie_count` <0 to determine if there no results.

https://yts.mx/api/v2/list_movies.json?sort=date_added&limit=50&query_term=Land+2020 also returns `"movie_count":49`, which is the number of results for `Land`, so it then filters by the year, and if there are 0 results it breaks.

**Comment by XYZJR** [comment]
So if `"The"` has results (1000+), then it checks if `"The+Matrix"` has its own results (4), and then it tests `"The+Matrix+1970"` which would still show the previous word combination's result of 4?

If it works that way, shouldn't we just **remove the error log**?

**Comment by ilike2burnthing** [comment]
Possibly. Not sure if there's a better approach. Will wait for ngosang to have a look.

File: src/Jackett.Common/Indexers/YTS.cs [hunk 1]
```diff
@@ -111,107 +110,126 @@ public async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query, int
...
-                    return releases.ToArray();
+                    return new List<ReleaseInfo>();
                 }
 
-                var movies = data_items.Value<JToken>("movies");
+                var movies = dataItems.Value<JToken>("movies");
                 if (movies == null)
-                    throw new Exception("API error, movies missing");
+                {
+                    logger.Error($"YTS API Error: Expected {movieCount} movie(s) in result, but found 0.");
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/YTS.cs:144]: Remove the error message.

**Review by ngosang — APPROVED** [review]
Thank you. LGTM

**Pull request merged**